torch/_inductor/scheduler.py

```diff
@@ -1724,17 +1702,18 @@ def add_user(
                     # this node must run after the prior writer
                     add_user(alt_name, node)
                     node.add_fake_dep(StarDep(alt_name, mode=node_mode))
-                    known_dep_node_names = dep_closure(node)
                     for user in name_to_users[alt_name].items:
+                        if user.get_name() == node.get_name():
+                            continue
+
                         assert isinstance(user.node, BaseSchedulerNode)
                         for other_name in user.node.get_buffer_names():
                             # this node must run after all prior readers
                             other_name = rename(other_name)
-                            if other_name not in known_dep_node_names:
```

Comment:
Why remove this? should be a good optimization to not add extra deps

Reply:
It's not sound. Say we have:
```python
def fn(x):
    y = x[::-1].clone()
    x.copy_(y + 1)
    return y, x
```
y is in the dep closue of the copy node, but it isn't an in-placeable use of x. This is why we have the extra fixup in `can_fuse_vertical`